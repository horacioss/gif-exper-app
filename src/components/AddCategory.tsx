import React, { Dispatch, SetStateAction, useState } from "react";

export interface AddCategoryProps {
  setCategories: Dispatch<SetStateAction<string[]>>;
}

const AddCategory = ({ setCategories }: AddCategoryProps) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value);
  };

  const hanleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={hanleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

export default AddCategory;
