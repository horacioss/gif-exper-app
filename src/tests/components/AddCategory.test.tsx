import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("tests over <AddCategory />", () => {
  const setCategories = jest.fn();
  const value = "Samurai X";
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("should render component Correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should change the value of input", () => {
    const input = wrapper.find("input");

    input.simulate("change", { target: { value } });
  });

  test("should not submit data from onSubmit...", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("should call setCategories() & clear text box", () => {
    const input = wrapper.find("input");
    input.simulate("change", { target: { value } });

    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    expect(input.prop("value")).toBe("");
  });
});
