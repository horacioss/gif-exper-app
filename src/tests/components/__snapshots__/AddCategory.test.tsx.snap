// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests over <AddCategory /> should render component Correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
